### PR TITLE
Add support for parsing #COMPARE function from lucene query

### PR DESCRIPTION
**Pull request by apmoriarty**
Addresses #1332

### CODE REVIEW

**Comment by d-hwang** [comment]
Would you be able to update web-services/deploy/docs/docs/query_help.html with some examples for both syntax? I missed updating it when I worked on it. Thanks!

**Review by d-hwang — DISMISSED** [review]

**Comment by Gara312** [comment]
> Сможете ли вы обновить web-services / deploy / docs / docs / query_help.html некоторыми примерами для обоих синтаксисов? Я пропустил обновление, когда работал над этим. Спасибо!

**Review by FineAndDandy — DISMISSED** [review]

**Review by keith-ratcliffe — COMMENTED** [review]
LGTM

**Review by keith-ratcliffe — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/EvaluationPhaseFilterFunctions.java [hunk 1]
```diff
@@ -1643,7 +1643,9 @@ public static boolean compare(Object field, String operator, String compareMode,
             }
         }
         
-        return compareFields(fields, operator, CompareFunctionValidator.Mode.valueOf(compareMode) == CompareFunctionValidator.Mode.ANY, fields2);
+        boolean matchesAny = compareMode.equalsIgnoreCase("any");
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/functions/EvaluationPhaseFilterFunctions.java]: why can't we use the enum here for comparison?

File: warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Compare.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * Function args are a field name, an operator, a mode, and another field name.
+ * <p>
+ * Lucene Example: #COMPARE(FIELD_A, &lt;, ANY, FIELD_B)
+ * <p>
+ * Jexl Example: filter:compare(FIELD_A, &lt;, ANY, FIELD_B)
+ */
+public class Compare extends AbstractEvaluationPhaseFunction {
+    
+    private final Set<String> validOpArgs = Sets.newHashSet("<", "<=", ">", ">=", "==", "=", "!=");
+    private final Set<String> validModeArgs = Sets.newHashSet("ANY", "ALL");
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Compare.java]: Why can't we use the enum names here?

File: warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Compare.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    
+    public Compare() {
+        super("compare");
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Compare.java:20]: How about a constant for "compare".  I see it used below as well.

**Review by ivakegg — CHANGES_REQUESTED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/EvaluationPhaseFilterFunctions.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/jexl/functions/EvaluationPhaseFilterFunctions.java]: fixed in next commit

File: warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Compare.java (hunk 3 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Compare.java:20]: As-is this follows the pattern of the include/exclude functions. I can extract that to a constant if you prefer

File: warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Compare.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Compare.java]: good call, that will be in the next commit

File: warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Compare.java (hunk 3 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Compare.java:20]: its fine as is

**Review by ivakegg — APPROVED** [review]

**Review by d-hwang — APPROVED** [review]

**Pull request merged**